compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/test/linalg_function_outlining.mlir

```diff
@@ -72,88 +72,38 @@ func.func @matmul_example(%A: memref<4x8xbf16>, %B: memref<8x4xbf16>, %C: memref
 
 // -----
 
-// Test demonstrating different kind of elementwise operations being mapped to a
+
+// Test demonstrating different kind of matmul operations being mapped to a
 // unique corresponding outlined function.
 
-// CHECK-LABEL: func.func private @generic_elementwise_1_outlined
-// CHECK-SAME:  (%[[INPUT:.*]]: memref<4xf32>,
-// CHECK-SAME:   %[[OUTPUT:.*]]: memref<4xbf16>) {
-// CHECK:           linalg.generic
-// CHECK-SAME:          ins(%[[INPUT]] :
-// CHECK-SAME:          outs(%[[OUTPUT]] :
-// CHECK:           arith.truncf
-// CHECK:           arith.addf
-// CHECK:           return
-// CHECK:        }
-// CHECK:       func.func private @generic_elementwise_0_outlined
-// CHECK-SAME:  (%[[INPUT:.*]]: memref<4xf32>,
-// CHECK-SAME:   %[[OUTPUT:.*]]: memref<4xbf16>) {
-// CHECK:           linalg.generic
-// CHECK-SAME:          ins(%[[INPUT]] :
-// CHECK-SAME:          outs(%[[OUTPUT]] :
-// CHECK:           arith.truncf
-// CHECK:           return
-// CHECK:        }
-// CHECK-LABEL:  func.func @elemwise_example
-// CHECK-SAME:   (%[[A:.*]]: memref<4xf32>,
-// CHECK-SAME:    %[[C:.*]]: memref<4xbf16>,
-// CHECK-SAME:    %[[B:.*]]: memref<4xf32>) {
-// CHECK:            amdaie.core
-// CHECK:               func.call @generic_elementwise_0_outlined(%[[A]], %[[C]])
-// CHECK-NOT:           linalg.generic
-// CHECK:               amdaie.end
-// CHECK:            }
-// CHECK:            amdaie.core
-// CHECK:               func.call @generic_elementwise_0_outlined(%[[B]], %[[C]])
-// CHECK-NOT:           linalg.generic
-// CHECK:               amdaie.end
-// CHECK:            }
-// CHECK:            amdaie.core
-// CHECK:               func.call @generic_elementwise_1_outlined(%[[A]], %[[C]])
-// CHECK-NOT:           linalg.generic
-// CHECK:               amdaie.end
-// CHECK:            }
-// CHECK:            return
-// CHECK:        }
-func.func @elemwise_example(%A: memref<4xf32>, %C: memref<4xbf16>, %B: memref<4xf32>) {
+// CHECK-DAG:  func.func private @[[MATMUL_K6:.*]]({{.*}}memref<4x6xbf16>, {{.*}}memref<6x4xbf16>, {{.*}}memref<4x4xf32>)
+// CHECK-DAG:  func.func private @[[MATMUL_K4:.*]]({{.*}}memref<4x4xbf16>, {{.*}}memref<4x4xbf16>, {{.*}}memref<4x4xf32>)
+// CHECK-NOT:  func.func private
+// CHECK:      func.func @matmul_example_2(
+// CHECK-SAME:  %[[A0:.*]]: memref<4x4xbf16>, %[[B0:.*]]: memref<4x4xbf16>,
+// CHECK-SAME:  %[[A1:.*]]: memref<4x6xbf16>, %[[B1:.*]]: memref<6x4xbf16>,
+// CHECK-SAME:   %[[C:.*]]: memref<4x4xf32>) {
+// CHECK:      amdaie.core
+// CHECK-NEXT: func.call @[[MATMUL_K4]](%[[A0]], %[[B0]], %[[C]])
+// CHECK-NEXT: func.call @[[MATMUL_K4]](%[[A0]], %[[B0]], %[[C]])
+// CHECK-NEXT: func.call @[[MATMUL_K4]](%[[B0]], %[[A0]], %[[C]])
+// CHECK-NEXT: func.call @[[MATMUL_K6]](%[[A1]], %[[B1]], %[[C]])
+// CHECK-NEXT: amdaie.end
+// CHECK:      return
+func.func @matmul_example_2(%A0: memref<4x4xbf16>, %B0: memref<4x4xbf16>,
```

Comment:
No need for this example. If required, you can add one `linalg.matmul` to an already existing test case above.

Reply:
I like this test, I think it's good to have a sequence of tests which slowly change what's being tested from the preceding test, makes diagnosing failures easier. I should use a more descriptive name though...